crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -124,8 +124,11 @@ use std::{
 ///
 /// # Safety
 ///
-/// The implementor must ensure that [`SystemParam::init_state`] correctly registers all
-/// [`World`] accesses used by this [`SystemParam`] with the provided [`system_meta`](SystemMeta).
+/// The implementor must ensure the following is true.
+/// - [`SystemParam::init_state`] correctly registers all [`World`] accesses used
+///   by this [`SystemParam`] with the provided [`system_meta`](SystemMeta).
+/// - None of the world accesses may conflict with any prior accesses registered
+///   on `system_meta`.
```

Comment:
The safety invariants for this were (rightfully) updated. Can you update the safety comments for implementors of the trait?